pkg/image/source_test.go

```diff
@@ -282,82 +292,81 @@ func TestParseScheme(t *testing.T) {
 func TestDetectSourceFromPath(t *testing.T) {
 	tests := []struct {
 		name           string
-		paths          []string
+		path           string
+		getFS          func(*testing.T, afero.Fs)
 		expectedSource Source
-		sourceType     string
 		expectedErr    bool
 	}{
 		{
 			name:           "no tar paths",
-			paths:          []string{},
-			sourceType:     "tar",
+			path:           "image.tar",
+			getFS:          getDummyTar("image.tar"),
```

Comment:
I'm not sure it can be avoided for all test cases. For example, in the `no path given` test case, `path` is set to `"/does-not-exist"` even though the `afero.Fs` is empty.